### PR TITLE
Cisdp 1815

### PR DESCRIPTION
**Pull request by slamothe99**
## Pre-review checklist

I have ensured that:

- [ ] Code runs successfully in the development environment.
- [ ] New code is tested to prove that it works. If not, justify reasoning.
- [ ] New code is documented using [type hinting](https://realpython.com/lessons/type-hinting/) and following the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) docstring format.
- [ ] New code follows the project naming standards.

## Post-review checklist

As I merge these changes, I have:

- [ ] Updated the related Jira ticket, including approximate time spent (including review).
- [ ] Moved the related Jira ticket to Done.

### CODE REVIEW

File: cishouseholds/edit.py [hunk 1]
```diff
@@ -10,6 +11,36 @@
...
+    infant_column_pattern_with_exceptions: str,
+    participant_column_pattern: str,
+    non_consented_count: str,
+):
+    """ """
+    infant_columns = [x for x in df.columns if re.match(infant_column_pattern, x)]
+    infant_columns_with_exceptions = [x for x in df.columns if re.match(infant_column_pattern_with_exceptions, x)]
+    participant_columns = [x for x in df.columns if re.match(participant_column_pattern, x)]
+
+    df = df.withColumn("INFANT_ARRAY", F.array([F.when(F.col(col).isNull(), 0).otherwise(1) for col in infant_columns]))
```
Comment (foster999) [inline cishouseholds/edit.py:27]: Please change these columns to objects, rather than creating columns that need dropping later

Comment (foster999) [reply cishouseholds/edit.py:27]: Also this code and test belongs in a separate PR. Please create this

Comment (foster999) [reply cishouseholds/edit.py:27]: Also, this should have replaced an existing function, please make sure that is removed. And appropriate function calls are made

File: cishouseholds/pipeline/load.py [hunk 2]
```diff
@@ -177,14 +177,19 @@ def update_processed_file_log(df: DataFrame, filename_column: str, file_type: st
     """Collects a list of unique filenames that have been processed and writes them to the specified table."""
     spark_session = get_or_create_spark_session()
     newly_processed_files = df.select(filename_column).distinct().rdd.flatMap(lambda x: x).collect()
+    file_lengths = df.groupBy(filename_column).count().select("count").rdd.flatMap(lambda x: x).collect()
```
Comment (foster999) [inline cishouseholds/pipeline/load.py:180]: This is a count of filenames. We need to get a count of the rows within each file here

**Review by foster999 — CHANGES_REQUESTED** [review]

File: cishouseholds/pipeline/load.py (hunk 2 above)
Comment (slamothe99) [reply cishouseholds/pipeline/load.py:180]: every row from the input file has a cell that contains the filename from which it originated when its added to the main dataframe therefore a count of the instances of that filename gives us the number of rows in the original file as df represents the complete df

**Review by foster999 — APPROVED** [review]

**Pull request merged**